plugins/chat/app/controllers/api/chat_channels_archives_controller.rb

```diff
@@ -8,35 +8,49 @@ def create
       guardian.ensure_can_change_channel_status!(channel_from_params, :archived)
       raise Discourse::InvalidAccess if !existing_archive.failed?
       Chat::ChatChannelArchiveService.retry_archive_process(chat_channel: channel_from_params)
-    else
-      archive_params =
-        params
-          .require(:archive)
-          .tap do |ca|
-            ca.require(:type)
-            ca.permit(:title, :topic_id, :category_id, tags: [])
-          end
-
-      new_topic = archive_params[:type] == "new_topic"
-      raise Discourse::InvalidParameters if new_topic && archive_params[:title].blank?
-      raise Discourse::InvalidParameters if !new_topic && archive_params[:topic_id].blank?
-
-      if !guardian.can_change_channel_status?(channel_from_params, :read_only)
-        raise Discourse::InvalidAccess.new(I18n.t("chat.errors.channel_cannot_be_archived"))
-      end
-
-      Chat::ChatChannelArchiveService.begin_archive_process(
-        chat_channel: channel_from_params,
-        acting_user: current_user,
-        topic_params: {
-          topic_id: archive_params[:topic_id],
-          topic_title: archive_params[:title],
-          category_id: archive_params[:category_id],
-          tags: archive_params[:tags],
-        },
-      )
+      return render json: success_json
     end
 
+    new_topic = archive_params[:type] == "new_topic"
+    raise Discourse::InvalidParameters if new_topic && archive_params[:title].blank?
+    raise Discourse::InvalidParameters if !new_topic && archive_params[:topic_id].blank?
+
+    if !guardian.can_change_channel_status?(channel_from_params, :read_only)
+      raise Discourse::InvalidAccess.new(I18n.t("chat.errors.channel_cannot_be_archived"))
+    end
+
+    if new_topic
+      valid, errors = Chat::ChatChannelArchiveService.validate_topic_params(guardian, topic_params)
+      return render json: failed_json.merge(errors: errors), status: 400 if !valid
+    end
+
+    Chat::ChatChannelArchiveService.begin_archive_process(
+      chat_channel: channel_from_params,
+      acting_user: current_user,
+      topic_params: topic_params,
+    )
```

Comment:
I feel like all this logic should be part of `Chat::ChatChannelArchiveService` which could return an error state

Reply:
> I feel like all this logic should be part of Chat::ChatChannelArchiveService.begin_archive_process which could return an error state

Yes I agree actually, will move it

> Minor note, I'm not sure the begin is necessary here.

Not sure what you mean by this? Would you prefer `start_archive_process` or something?

Reply:
I mean that potentially every function call is a `begin` :D even an `end` could be `begin_ending`. So yes maybe something more simple which doesn't imply theres the opposite too. When you have `start_xxx` I somehow expect to have `stop_xxx`.

Reply:
@jjaffeux I moved the logic into the class, not totally happy with it but I will return to it when we have nailed down more of an idea of how services will be laid out

Reply:
> I mean that potentially every function call is a `begin` :D even an `end` could be `begin_ending`. So yes maybe something more simple which doesn't imply theres the opposite too. When you have `start_xxx` I somehow expect to have `stop_xxx`.

I do see what you mean...personally I would not always expect there to be the opposite, in this case it starts here but ends somewhere else. I will rename to something else though, `create_archive_process` I think is safe?

Reply:
Yep looks good to me 👍